### DISCUSSION TITLE
能不能解析图片中的文字

### DISCUSSION THREAD

**Discussion by Tendo33** [magicpdf/Magic-PDF#795] [Ideas]
我想强制解析图片，开ocr 还是会识别成一个图片，并直接保存，有没有一种方法解析图片中的文字

**Comment by Potato-wll** [comment]
你自己做一个后处理，拿到图片后调以下PaddleOCR就可以了

**Reply by Tendo33** [reply]
能不能利用已经加载进GPU的模型啊

**Reply by Potato-wll** [reply]
> 能不能利用已经加载进GPU的模型啊

可以，里面有ocr模型，你从调用ocr的接口一路往里，就能看到是怎么调用的了，他用的就是paddleOCR的模型，你看看paddleOCR是咋加载推理的应该也没问题

**Reply by Tendo33** [reply]
最多只能改成`model_config.__model_mode__ = "lite"` 还是会后计算，再往里只能改源码了

**Comment by wcool1** [comment]
> 我想强制解析图片，开ocr 还是会识别成一个图片，并直接保存，有没有一种方法解析图片中的文字

兄弟，你后来是如何处理的？

**Reply by Potato-wll** [reply]
两个思路，第一个就是修改源码，我读过0.9.2版本的代码，作者的解析思路是基于layout，ocr，公式layout等模型，把相关区域的坐标划定出来，然后再根据区域的不同标签，走不同的逻辑，最后把所有的解析内容合并，你可以在根据区域标签解析的那一块，把你的ocr解析加进去。
第二个就是解析完会有模型的json结果，包含坐标标签，你可以遍历一下那个json，自己再实例化一个ocr模型去解析其中的图片，或者你直接遍历一下输出的markdown文件，遇到图片就读取图片内容，ocr提取文字。

**Reply by wcool1** [reply]
感谢您的回复，我想请问一下您是否根据该思路实现过？我更改了0.10相关逻辑的源码但是仍然无法识别，贡献者回复说是layout模型没有相关的训练 [链接](https://github.com/opendatalab/MinerU/issues/1415) ，另外，第二种思路：运行结果中只有一整张图片的位置，图片中内容也有布局坐标信息吗？如果有，我该如何调用OCR模型将识别后的文本和布局坐标进行结合输出呢？

**Reply by Tendo33** [reply]
我直接调用的 from magic_pdf.model.pp_structure_v2 import CustomPaddleModel，解析完图片之后把item["text"] 提出来就行了

**Reply by Potato-wll** [reply]
我没有理解你说的图片中的内容布局坐标信息，ocr识别，输入的是图片输出的是文字啊。而且不管你是基于坐标去原pdf中截取图片，还是通过遍历markdown，都是可以获取完整的图片的，把图片送入到PaddleOCR模型就可以得到图片中的文字信息。图片在pdf中的坐标可以参考https://github.com/opendatalab/MinerU/blob/master/docs/output_file_zh_cn.md获取,或者你遍历markdown，markdown中的图片都有特殊标识的，可以很好的get到。基于paddle ocr提取图片文字内容可参考以下代码：
from paddleocr import PaddleOCR
ocr_model = PaddleOCR()
image_text = ocr_model.ocr(image_path,cls=True)

**Comment by greatitman** [comment]
先利用paddle ocr识别出首页的语种，然后按解析的语种丢给mineru解析
